src/Microsoft.AspNetCore.Authorization/AuthorizationPolicy.cs

```diff
@@ -122,8 +122,14 @@ public static AuthorizationPolicy Combine(IEnumerable<AuthorizationPolicy> polic
 
             var policyBuilder = new AuthorizationPolicyBuilder();
             var any = false;
-            foreach (var authorizeAttribute in authorizeData.OfType<AuthorizeAttribute>())
+            foreach (var authorizeDatum in authorizeData)
```

Comment:
May be worth adding a comment to explain this is not using Linq for perf reason in order to avoid regression?